apps/judicial-system/web/src/utils/validate.ts

```diff
@@ -297,10 +302,65 @@ export const isProcessingStepValidIndictments = (
 export const isTrafficViolationStepValidIndictments = (
   workingCase: Case,
 ): boolean => {
-  return Boolean(
+  const hasValidDemands = Boolean(
     workingCase.demands &&
       (!workingCase.hasCivilClaims || workingCase.civilDemands),
   )
+
+  if (!workingCase.indictmentSubtypes || !hasValidDemands) {
+    return false
+  }
+
+  const validateIndictmentCount = (indictmentCount: IndictmentCount) =>
+    indictmentCount.policeCaseNumber &&
+    indictmentCount.offenses &&
+    indictmentCount.offenses?.length > 0 &&
+    indictmentCount.vehicleRegistrationNumber &&
+    indictmentCount.lawsBroken &&
+    indictmentCount.incidentDescription &&
+    indictmentCount.legalArguments
+
+  // All indictment counts are traffic violations
+  if (isTrafficViolationCase(workingCase)) {
+    const hasValidTrafficViolationIndictmentCounts =
+      workingCase.indictmentCounts?.every(validateIndictmentCount) ?? false
+
+    return hasValidTrafficViolationIndictmentCounts
+  }
+
+  // Some indictment counts are traffic violations while others are not
+  const [onlyTrafficViolationIndictmentCounts, otherIndictmentCounts] =
+    partition(workingCase.indictmentCounts, (indictmentCount) => {
+      const policeCaseNumber = indictmentCount.policeCaseNumber
+
+      return isTrafficViolationIndictmentCount(
+        policeCaseNumber,
+        workingCase.indictmentSubtypes,
```

Comment:
What if we have multiple subtypes here and one of them is `IndictmentSubtype.TRAFFIC_VIOLATION`, then `isTrafficViolationIndictmentCount` will return `false`, right? Is that expected here?

Reply:
This comment was made before a refactor was done on this code. Could you please take another look 👀🙏